pkg/ha/filter.go

```diff
@@ -208,33 +212,33 @@ func findDataTimeRange(tts []prompb.TimeSeries) (minTUnix int64, maxTUnix int64)
 	return minTUnix, maxTUnix
 }
 
-func haLabels(labels []prompb.Label) (cluster, replica string) {
+func (h *Filter) haLabels(labels []prompb.Label) (cluster, replica string) {
 	for _, label := range labels {
-		if label.Name == ClusterNameLabel {
+		if label.Name == h.clusterLabelName {
 			cluster = label.Value
-		} else if label.Name == ReplicaNameLabel {
+		} else if label.Name == h.replicaLabelName {
 			replica = label.Value
 		}
 	}
 	return cluster, replica
 }
 
-func validateClusterLabels(cluster, replica string) error {
+func (h *Filter) validateClusterLabels(cluster, replica string) error {
 	if cluster == "" && replica == "" {
 		return fmt.Errorf("HA enabled, but both %s and %s labels are empty",
-			ClusterNameLabel,
-			ReplicaNameLabel,
+			h.clusterLabelName,
+			h.replicaLabelName,
 		)
 	} else if cluster == "" {
 		return fmt.Errorf("HA enabled, but %s label is empty; %s set to: %s",
-			ClusterNameLabel,
-			ReplicaNameLabel,
+			h.clusterLabelName,
+			h.replicaLabelName,
 			replica,
 		)
 	} else if replica == "" {
```

Comment:
Should we also check `cluster == replica`?

Reply:
We can add, which also makes me think that we should convert this block of if-else-if to switch. It will be easier to read.

Reply:
I don't see a problem in `cluster` and `replica` label values being same. Someone can have external labels like 

```yaml
cluster: "one"
replica: "one"
---
cluster: "one"
replica: "two"
```

which should work perfectly fine.